IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -47,71 +47,132 @@ - (instancetype)init
     return self;
 }
 
-- (void)storeCredential:(MSIDCredentialCacheItem *)credential forKey:(MSIDCacheKey *)key
+- (void)storeItem:(id<MSIDJsonSerializable>)item forKey:(MSIDCacheKey *)key
 {
     dispatch_barrier_async(self.queue, ^{
-        [self.cacheObjects setObject:credential forKey:key];
+        NSString *type = [self getItemTypeFromCacheKey:key];
+        
+        if (type)
+        {
+            NSMutableDictionary *items = [self.cacheObjects objectForKey:type];
+            
+            if (!items)
+            {
+                items = [NSMutableDictionary new];
+            }
+            
+            [items setObject:item forKey:key];
+            [self.cacheObjects setObject:items forKey:type];
+        }
     });
 }
 
 - (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem
 {
     dispatch_barrier_async(self.queue, ^{
-        for (MSIDCacheKey *key in storageItem.cacheObjects)
+        
+        for (NSString *typeKey in storageItem.cacheObjects)
         {
-            MSIDCredentialCacheItem *credential = [storageItem.cacheObjects objectForKey:key];
-            if (credential)
+            NSMutableDictionary *typeDict = [storageItem.cacheObjects msidObjectForKey:typeKey ofClass:[NSDictionary class]];
+            NSMutableDictionary *subDict = [self.cacheObjects msidObjectForKey:typeKey ofClass:[NSDictionary class]];
+            
+            if (typeDict)
             {
-                [self.cacheObjects setObject:credential forKey:key];
+                if (!subDict)
+                {
+                    [self.cacheObjects setObject:typeDict forKey:typeKey];
+                }
+                else
+                {
+                    [subDict addEntriesFromDictionary:typeDict];
+                    [self.cacheObjects setObject:subDict forKey:typeKey];
+                }
             }
             else
             {
-                MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Credential is nil for key %@ while merging storage credentials.", MSID_PII_LOG_MASKABLE(key));
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to get type dictionary from key for stored credential.");
             }
         }
     });
 }
 
-- (void)removeStoredCredentialForKey:(MSIDCacheKey *)key
+- (void)removeStoredItemForKey:(MSIDCacheKey *)key
 {
     dispatch_barrier_sync(self.queue, ^{
```

Comment:
does this have to be `sync` instead of `async`?

Reply:
Good catch. Addressed.:)